include/timer.h

```diff
@@ -62,4 +80,61 @@ static inline void Delay(int milliseconds)
 	std::this_thread::sleep_for(std::chrono::milliseconds(milliseconds));
 }
 
+static inline void DelayUs(int microseconds)
+{
+	std::this_thread::sleep_for(std::chrono::microseconds(microseconds));
+}
+
+// The duration to use for precise sleep
+static constexpr int precise_delay_duration_us = 100;
+// The tolerance to allow for sleep variation
+static constexpr int precise_delay_tolerance_us = precise_delay_duration_us; 
+// The estimate of how long the sleep should take (microseconds)
+static constexpr double precise_delay_default_estimate = 5e-5;
+// Use the estimate value as the defaul mean time taken
+static constexpr double precise_delay_default_mean = 5e-5;
+
+static inline void DelayPrecise(int milliseconds)
```

Comment:
In function implementations, input args can be locked as `const` if we're not modifying them (even for simple scalar types like this).  It helps me as a reader know if this variable is locked-in or not when reading the code.